src/interpreter/mod.rs

```diff
@@ -1096,8 +1105,8 @@ mod tests {
             x_only_pks.push(x_only_pk);
             let schnorr_sig = secp.sign_schnorr_with_aux_rand(&msg, &keypair, &[0u8; 32]);
             let schnorr_sig = bitcoin::taproot::Signature {
-                sig: schnorr_sig,
-                hash_ty: bitcoin::sighash::TapSighashType::Default,
+                signature: schnorr_sig,
+                sighash_type: bitcoin::sighash::TapSighashType::Default,
             };
             ser_schnorr_sigs.push(schnorr_sig.to_vec());
```

Comment:
Can't we do `serialize` trick here to avoid one heap allocation?
Also aren't we intending to deprecate `to_vec` anyways?

Reply:
#687